poc/knn/AwakeData.py

```diff
@@ -1,62 +1,180 @@
-plt.ylabel('Pre#dicted')
-plt.show()
-
-#error = []
-
-# Calculating error for K values between 1 and 40
-#for i in range(1, 40):
-#    knn = KNeighborsClassifier(n_neighbors=i)
-#    knn.fit(X_train, y_train)
-#    pred_i = knn.predict(X_test)
-#    error.append(np.mean(pred_i != y_test))
-#   
-#plt.figure(figsize=(12, 6))
-#plt.plot(range(1, 40), error, color='red', linestyle='dashed', marker='o',
-#         markerfacecolor='blue', markersize=10)
-#plt.title('Error Rate K Value')
-#plt.xlabel('K Value')
-#plt.ylabel('Mean Error')
-
-#plt.show()
\ No newline at end of file
+
+def plotValidationModelCurves(estimator, title, X, y, axes=None, ylim=None, cv=None,
+                              n_jobs=None, train_sizes=np.linspace(.1, 1.0, 5)):
+    if axes is None:
+        _, axes = plt.subplots(1, 3, figsize=(20, 5))
+
+    axes[0].set_title(title)
+    if ylim is not None:
+        axes[0].set_ylim(*ylim)
+    axes[0].set_xlabel("Training samples")
+    axes[0].set_ylabel("Score")
+
+    train_sizes, train_scores, test_scores, fit_times, _ = \
+        learning_curve(estimator, X, y, cv=cv, n_jobs=n_jobs,
+                       train_sizes=train_sizes,
+                       return_times=True)
+    train_scores_mean = np.mean(train_scores, axis=1)
+    train_scores_std = np.std(train_scores, axis=1)
+    test_scores_mean = np.mean(test_scores, axis=1)
+    test_scores_std = np.std(test_scores, axis=1)
+    fit_times_mean = np.mean(fit_times, axis=1)
+    fit_times_std = np.std(fit_times, axis=1)
+
+    axes[0].grid()
+    axes[0].fill_between(train_sizes, train_scores_mean - train_scores_std,
+                         train_scores_mean + train_scores_std, alpha=0.1,
+                         color="r")
+    axes[0].fill_between(train_sizes, test_scores_mean - test_scores_std,
+                         test_scores_mean + test_scores_std, alpha=0.1,
+                         color="g")
+    axes[0].plot(train_sizes, train_scores_mean, 'o-', color="r",
+                 label="Training score")
+    axes[0].plot(train_sizes, test_scores_mean, 'o-', color="g",
+                 label="Cross-validation score")
+    axes[0].legend(loc="best")
+
+    axes[1].grid()
+    axes[1].plot(train_sizes, fit_times_mean, 'o-')
+    axes[1].fill_between(train_sizes, fit_times_mean - fit_times_std,
+                         fit_times_mean + fit_times_std, alpha=0.1)
+    axes[1].set_xlabel("Training examples")
+    axes[1].set_ylabel("fit_times")
+    axes[1].set_title("Scalability of the model")
+
+    axes[2].grid()
+    axes[2].plot(fit_times_mean, test_scores_mean, 'o-')
+    axes[2].fill_between(fit_times_mean, test_scores_mean - test_scores_std,
+                         test_scores_mean + test_scores_std, alpha=0.1)
+    axes[2].set_xlabel("fit_times")
+    axes[2].set_ylabel("Score")
+    axes[2].set_title("Performance of the model")
+
+
+def main():
+    dataset = pd.read_csv("../../../SleepAnalyzer/outputAwakeInformation/sleepLogs.csv")
```

Comment:
Shouldn't this be sent by the user on script execution instead of it being a **static path?**